File: compatibility-tests/src/test/java/nnl/rocks/kactoos/text/TrimmedTextTest.java
/**
 * The MIT License (MIT)
 * <p>
 * Copyright (c) 2017-2018 Yegor Bugayenko
 * <p>
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 * <p>
 * The above copyright notice and this permission notice shall be included
 * in all copies or substantial portions of the Software.
 * <p>
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */
package nnl.rocks.kactoos.text;

import nnl.rocks.kactoos.matchers.TextHasString;
import org.hamcrest.MatcherAssert;
import org.junit.Test;

/**
 * Test case for {@link TrimmedText}.
 * @author Vseslav Sekorin (vssekorin@gmail.com)
 * @version $Id: 0858fe3a68709b7f6e2eb6059d79e6f06cd00ab6 $
 * @since 0.1
 * @checkstyle JavadocMethodCheck (500 lines)
 */
public final class TrimmedTextTest {

    @Test
    public void convertsText() {
        MatcherAssert.assertThat(
            "Can't trim a text",
            new TrimmedText(new TextOf("  Hello!   \t ")),
            new TextHasString("Hello!")
        );
    }

    @Test
    public void trimmedBlankTextIsEmptyText() {
        MatcherAssert.assertThat(
            "Can't trim a blank text",
            new TrimmedText(new TextOf("  \t ")),
            new TextHasString("")
        );
    }
}

File: kactoos-jvm/src/test/kotlin/nnl/rocks/kactoos/text/TrimmedTextTest.kt

package nnl.rocks.kactoos.text

import nnl.rocks.kactoos.matchers.TextHasString
import org.hamcrest.MatcherAssert
import org.junit.Test

/**
 * Test case for [TrimmedText].
 * @author Vseslav Sekorin (vssekorin@gmail.com)
 * @version $Id: 0858fe3a68709b7f6e2eb6059d79e6f06cd00ab6 $
 * @since 0.1
 * @checkstyle JavadocMethodCheck (500 lines)
 */
class TrimmedTextTest {

    @Test
    fun convertsText() {
        MatcherAssert.assertThat(
                "Can't trim a text",
                TrimmedText(TextOf("  Hello!   \t ")),
                TextHasString("Hello!")
        )
    }

    @Test
    fun trimmedBlankTextIsEmptyText() {
        MatcherAssert.assertThat(
                "Can't trim a blank text",
                TrimmedText(TextOf("  \t ")),
                TextHasString("")
        )
    }
}
